add getGenres support

Duncaen opened the pull request:
I never used gomigrate, so I'm not sure if this is correct.

This also adds a decoded name column to the genres table, I thought there was a indexed genre but there isn't I can remove the column if you think its not necessary.

closes #34

sentriz commented:
hi! thank you very much for this - looks very good. 
as far as i can remember the "u dec" columns are only used for searching (so you can look up artist/tracks/albums with non latin chars with just latin char equivalents) so i think it's safe to remove them for genres

https://github.com/sentriz/gonic/blob/master/server/ctrlsubsonic/handlers_by_tags.go#L157-L210

Duncaen commented:
I removed the decoded field from the genre table, added a `getSongsByGenre` view and added genre parameter support to the `getRandomSongs` view.

One additional thing I could add is add a genre parameter to `spec.NewTrackByTags` so that all the views that return `TrackChild` can fill in the genre field.

sentriz commented:
cool thank you :) i can have a proper look and merge when my next college work "sprint" ends. shouldn't be long

Duncaen commented:
Ok nice, I rebased on master and squashed the fixup commits, this is ready for review.

sentriz commented on server/ctrlsubsonic/handlers_common.go:
```diff
@@ -288,18 +288,27 @@ func (c *Controller) ServeGetSong(r *http.Request) *spec.Response {
 
 func (c *Controller) ServeGetRandomSongs(r *http.Request) *spec.Response {
```
thanks for fixing my `year` laziness here :)

sentriz commented:
sorry for the delay. nothing much i would change here. thank you for the feature :)
